core/models/access_token.py

```diff
@@ -0,0 +1,24 @@
+from django.db import models
+from django.utils import timezone
+
+from django_cyverse_auth.models import Token
+
+
+class AccessToken(models.Model):
+    """
+    Extend the django_cyverse_auth Token to add a name
+    """
+    token = models.OneToOneField(Token, on_delete=models.CASCADE)
+    name = models.CharField(max_length=128, null=False, blank=False)
+
+    class Meta:
+        db_table = "access_token"
+        app_label = "core"
+
+def create_access_token(user, token_name=None, token_expire=None, remote_ip=None, issuer=None):
+    if AccessToken.objects.filter(name=token_name, token__user_id=user.id):
+        return None
+    token = Token(user=user, issuer=issuer)
```

Comment:
This line and one below can be replaced by a single ```token = Token.objects.create(...)```

Reply:
Would this still check if a token with that name already exists? Or does it rely on the database enforcing that constraint?

Reply:
I'm referring to line 21, 22 here

Reply:
Oh okay. To make the name enforced by the database, I can probably add `unique=True` to the CharField and then I don't need 19 and 20 either

Reply:
That would mean make the name unique across all tokens, so two users couldn't have token 'foobar'